Added release notes for C++ (#714)
Azure/azure-sdk-for-cpp: sdk/core/azure-core/CHANGELOG.md

# Release History

## 1.0.0-beta.2 (2020-10-08)

### Breaking Changes

- Throw Azure::Http::TransportException if creating new connection fails.

> - Response objects store Nullable<T>.

Reviewer: Angle brackets don't render in markdown
```suggestion
- Response objects store Nullable\<T\>.
```
![image](https://user-images.githubusercontent.com/6527137/95532656-d730aa80-0995-11eb-9084-bf7489ae886e.png)

Author: Nice catch, fixed

### Bug Fixes

> - Switched to a more stable wait on sockets to address connection timeouts.

Reviewer: Just double checking, does this correspond to https://github.com/Azure/azure-sdk-for-cpp/pull/676?

Author: Yes

# Release History

> ## 1.0.0-beta.2 (2020-10-08)

Reviewer: What commits/code changes are included as part of this release (between beta.1 and 2)?

Author: I picked up changes since the last release on Sept 7th.  I didn't add mentions to the EngSys or non-public changes

Author: @ahsonkhan waiting on your review

Reviewer: Done. Thanks for the reminder.

> ### New Features
> 
> - Added NOTICE.txt for 3rd party dependencies
> 

Reviewer: This isn't really a **feature** of the SDK though from an API usage perspective. Remove (or move to "other changes" if you really want to keep it).
```suggestion
```

### Breaking Changes

> - Throw Azure::Http::TransportException if creating new connection fails

Reviewer: Be consistent and add a period.

```suggestion
- Throw Azure::Http::TransportException if creating new connection fails.
```

> - Response objects store Nullable<T>

Reviewer: ```suggestion
- Response objects store Nullable<T>.
```

### Bug Fixes

> - Switched to a more stable wait on sockets to address connection timeouts

Reviewer: ```suggestion
- Switched to a more stable wait on sockets to address connection timeouts.
```

> - Replace `Nullable(const T&)` with `Nullable(T)` to avoid extra copy when initialized with an rvalue.

Reviewer: ```suggestion
- Replace `Nullable(const T&)` with `Nullable(T)` to avoid extra copy when initialized with an r-value.
```

Author: C++ refers to them by rvalue, lvalue.  Its uncommon to use hyphen syntax.

### Other changes and Improvements

> - Improved performance on windows when using libcurl

Reviewer: ```suggestion
- Improved performance on windows when using libcurl.
```

> - Pinned the version of dependent packages

Reviewer: ```suggestion
- Pinned the version of package dependencies.
```
